tests/subsys/settings/functional/src/settings_basic_test.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2018 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ *  @brief Settings functional test suite
+ *
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <errno.h>
+#include <settings/settings.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(settings_basic_test);
+
+/*
+ * Test the two support routines that settings provides:
+ *
+ *   settings_name_steq(name, key, next): compares the start of name with key
+ *   settings_name_next(name, next): returns the location of the first
+ *                                   separator
+ */
+
+static void test_support_rtn(void)
+{
+	const char test1[] = "bt/a/b/c/d";
+	const char test2[] = "bt/a/b/c/d=";
+	const char *next1, *next2;
+	int rc;
+
+	/* complete match: return 1, next = NULL */
+	rc = settings_name_steq(test1, "bt/a/b/c/d", &next1);
+	zassert_true(rc == 1, "_steq comparison failure");
+	zassert_true(next1 == NULL, "_steq comparison next error");
+	rc = settings_name_steq(test2, "bt/a/b/c/d", &next2);
+	zassert_true(rc == 1, "_steq comparison failure");
+	zassert_true(next2 == NULL, "_steq comparison next error");
+
+	/* partial match: return 1, next <> NULL */
+	rc = settings_name_steq(test1, "bt/a/b/c", &next1);
+	zassert_true(rc == 1, "_steq comparison failure");
+	zassert_false(next1 == NULL, "_steq comparison next error");
+	rc = settings_name_steq(test2, "bt/a/b/c", &next2);
+	zassert_true(rc == 1, "_steq comparison failure");
+	zassert_false(next2 == NULL, "_steq comparison next error");
+
+	/* no match: return 0, next = NULL */
+	rc = settings_name_steq(test1, "bta", &next1);
+	zassert_true(rc == 0, "_steq comparison failure");
+	zassert_true(next1 == NULL, "_steq comparison next error");
+	rc = settings_name_steq(test2, "bta", &next2);
+	zassert_true(rc == 0, "_steq comparison failure");
+	zassert_true(next2 == NULL, "_steq comparison next error");
```

Comment:
What I am missing here is test1 against "b" - where we test if "bt/a/b/c/d" would (wrongly) show that it begins with "b" - in a string comparison, yes it is. But it should not match as "b" is not the same like "bt".

Reply:
I added this test and it does not match in case of "b", but this got me thinking: it should match "b", as this would allow to select all handlers that start with "b". The next value should then be "t/a/b...".

Reply:
No, it should not. It leads to a lot of confusions:  See the example that I am always providing here:
```bt``` from Bluetooth and ```btn``` from a button. What if you have ```bt/n``` key and ```btn```.

I really prefer it the way you have implemented it up to now.

Reply:
OK, I have reset to the old method.